### PR TITLE
fix: asym/sym logic

### PR DESCRIPTION
**Pull request by 0xApotheosis**
## Description

Fixes asym quote estimation logic so pool share and slippage is correct for both sym and asym deposits.

## Pull Request Type

- [x] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [ ] :nail_care: New Feature (Breaking/Non-breaking Change)

## Issue (if applicable)

Contributes to https://github.com/shapeshift/web/issues/6019

## Risk

Small - isolated to code paths behind the LP feature flag.

## Testing

Ensure both sym and asym LPs are similar to THOSwap.

### Engineering

☝️

### Operations

Not required at this stage.

## Screenshots (if applicable)

N/A

### CODE REVIEW

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 1]
```diff
@@ -257,18 +257,31 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
   useEffect(() => {
     ;(async () => {
       if (!runeCryptoLiquidityAmount || !assetCryptoLiquidityAmount || !asset) return
+      const isAsym = foundPool?.asymSide !== null
+      const isAsymAssetSide = foundPool?.asymSide === AsymSide.Asset
+      const isAsymRuneSide = foundPool?.asymSide === AsymSide.Rune
+
+      // If the pool is asymmetrical the then we are only depositing one asset amount, and the other is zero
```
Comment (gomesalexandre) [inline src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx:264]:
```suggestion
      // If the pool is asymmetrical then we are only depositing one asset amount, and the other is zero
```

**Review by gomesalexandre — CHANGES_REQUESTED** [review]
Tested locally:

### Asym asset side

image
image

Pool share is now *4 the THORSwap amount 🤯 

### Asym RUNE side

 
image
image

- This is also around *4 the TS pool share amount

### Symmetrical

image

- Rebalanced amounts look good, but the pool share is around 3ish times higher than THORSwap

image

**Comment by 0xApotheosis** [comment]
Confirming this got rugged when develop was merged in, will investigate and fix 👌

**Comment by 0xApotheosis** [comment]
After more investigation, I think @gomesalexandre you might have been checking against BNB on Beacon Chain instead of Smart Chain when using THORSwap (they use different colored asset icons than us).

I'm unable to get a delta from their values.

**Review by gomesalexandre — APPROVED** [review]
Tested again, confirmed I was a derp in my previous pass

### Asym asset

Screenshot 2024-01-19 at 09 07 07
Screenshot 2024-01-19 at 09 07 01



### Asset RUNE

Screenshot 2024-01-19 at 09 07 35
Screenshot 2024-01-19 at 09 07 42

### Sym

Screenshot 2024-01-19 at 09 08 24
Screenshot 2024-01-19 at 09 08 15

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx (hunk 1 above)
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: a595a9c1aa14d5463d075e1000711222f0064546

File: src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx [hunk 2]
```diff
@@ -257,18 +257,31 @@ export const AddLiquidityInput: React.FC<AddLiquidityInputProps> = ({
   useEffect(() => {
     ;(async () => {
       if (!runeCryptoLiquidityAmount || !assetCryptoLiquidityAmount || !asset) return
+      const isAsym = foundPool?.asymSide !== null
```
[the remark this answers is not in the document]
Comment (gomesalexandre) [reply src/pages/ThorChainLP/components/AddLiquitity/AddLiquidityInput.tsx]: a595a9c1aa14d5463d075e1000711222f0064546

**Pull request merged**